### PR TITLE
Break out owned-by worklist and add clinician filter

### PR DESCRIPTION
**Pull request by luketlancaster**
- [x] Add the Clubhouse Story ID: [ch22141]

### CODE REVIEW

**Comment by luketlancaster** [comment]
there's like a 100% chance I missed something, but it was time to get another set of eyes on this beast before I went totally crazy

File: src/js/views/patients/worklist/worklist_views.js
Comment (paulfalgout) [inline src/js/views/patients/worklist/worklist_views.js]: I think we're doing this too often in the code base.. we need this if we're doing something with two data points like `Nurse (NUR)` or whatever.. but in this case `attr: 'name'` should be enough

File: src/js/apps/patients/worklist/worklist-owned-by_app.js
Comment (paulfalgout) [inline src/js/apps/patients/worklist/worklist-owned-by_app.js]: let's make this line `const filter = this._filtersBy(this.getState('groupId'), this.getState('clinicianId'));`

File: src/js/apps/patients/worklist/worklist-owned-by_app.js
Comment (paulfalgout) [inline src/js/apps/patients/worklist/worklist-owned-by_app.js]: unused

File: src/js/apps/patients/worklist/worklist-owned-by_app.js
Comment (paulfalgout) [inline src/js/apps/patients/worklist/worklist-owned-by_app.js:50]: this happens in the `if` and the `else`  let's just make it the first line

File: src/js/apps/patients/worklist/worklist-owned-by_app.js
Comment (paulfalgout) [inline src/js/apps/patients/worklist/worklist-owned-by_app.js]: let's move this start into a function and call it in `onBeforeStart` along with the `listenTo`.  Also with changing the childApp above we won't need to set region or clinician here.  Then we should not need to set defaults with the `||`  if we do then the defaults aren't working above.

File: src/js/apps/patients/worklist/worklist-owned-by_app.js
Comment (paulfalgout) [inline src/js/apps/patients/worklist/worklist-owned-by_app.js]: let's change this to:
```
  filters: {
    AppClass: FiltersApp,
    regionName: 'filters',
    getOptions: ['currentClinician'],
  }
```

File: src/js/apps/patients/worklist/worklist-owned-by_app.js
Comment (paulfalgout) [inline src/js/apps/patients/worklist/worklist-owned-by_app.js]: you _could_ simplify this a bit I think.. I don't think it'd be too brittle, but this could be:
```js
this.listenTo(filtersApp, 'change:state', state = > {
  this.setState(state);
});
```
then in the change event's you'll be triggering `{ groupId: id }` etc

File: src/js/apps/patients/worklist/worklist-owned-by_app.js
Comment (paulfalgout) [inline src/js/apps/patients/worklist/worklist-owned-by_app.js]: we can update the other app too, but I don't see any reason we couldn't use `fetch:${ this.worklistType }:collection`

File: src/js/apps/patients/worklist/filters_app.js
Comment (paulfalgout) [inline src/js/apps/patients/worklist/filters_app.js]: probably a decent case for `this.mergeOptions(options, ['currentClinician', 'groupId', 'clinicianId']);`

File: src/js/apps/patients/worklist/filters_app.js
Comment (paulfalgout) [inline src/js/apps/patients/worklist/filters_app.js]: let's change this to:
```
const groups = this.currentClinician.getGroups();
if(groups.length < 2) return;

const groupsSelect = new GroupsDropList({
  collection: this._getGroups(groups),
```

**Review by paulfalgout — CHANGES_REQUESTED** [review]

File: src/js/apps/patients/worklist/worklist-owned-by_app.js
Comment (luketlancaster) [reply src/js/apps/patients/worklist/worklist-owned-by_app.js]: so if I move this to `onBeforeStart` the filters only work one time, then they don't seem to be changing the state at all

Comment (paulfalgout) [reply src/js/apps/patients/worklist/worklist-owned-by_app.js]: ah.. add `restartWithParent: false` to the childApp definition with `AppClass`

File: src/js/apps/patients/worklist/filters_app.js
Comment (paulfalgout) [inline src/js/apps/patients/worklist/filters_app.js]: since there's no async here all of the above can just go in `onStart`

File: src/js/views/patients/worklist/worklist_views.js
Comment (paulfalgout) [inline src/js/views/patients/worklist/worklist_views.js]: this doesn't need to be a function

File: src/js/apps/patients/worklist/worklist-owned-by_app.js
Comment (paulfalgout) [inline src/js/apps/patients/worklist/worklist-owned-by_app.js]: ya know.. 100% of this is view related.    Why don't we move `sortOptions` to `worklist_views` for now.

**Review by paulfalgout — CHANGES_REQUESTED** [review]

File: src/js/apps/patients/worklist/worklist-owned-by_app.js
Comment (luketlancaster) [reply src/js/apps/patients/worklist/worklist-owned-by_app.js:50]: yeah I was running into an issue with this...lemme see if I can't reproduce it

Comment (luketlancaster) [reply src/js/apps/patients/worklist/worklist-owned-by_app.js:50]: yup, so if I just make `this.getRegion('list')` the first line, it errors because the region is undefined. If keep only the bottom one, I get a "TypeError: pattern is undefined"

Comment (paulfalgout) [reply src/js/apps/patients/worklist/worklist-owned-by_app.js:50]: ah right.. because it's based off the layout.. carry on

**Review by paulfalgout — CHANGES_REQUESTED** [review]
Ok.. so.. after thinking about it.. one more iteration.. what if we moved calling `startFiltersApp` to after `showSortView` then in the method do `listenTo(filtersApp.getState(), 'change', state => { this.setState(state.attributes`  then make the app `restartWithParent: false`  and up in the filters app change the `triggerMethod`s to `setState`s  and instead of `mergeOptions` pass `state: this.getState().attributes }` which'll initiate the state model.  Then the only thing that needs to be added is:
```
stateEvents: {
  'change:clinicianId': 'showResetButton',
},
```

**Comment by paulfalgout** [comment]
Looks like we lost some coverage https://codecov.io/gh/RoundingWell/care-ops-frontend/compare/92b270f0eab480ef0e792ee63f2e92eb32b7e9cc...f91f5dd0918e4ec339a3753e3204da00e6c1bac9/src/src/js/apps/patients/worklist/worklist_app.js  need to add a couple of tests that are now only done on the owned-by page

**Review by paulfalgout — APPROVED** [review]
Approved once line coverage is 100%

**Comment by luketlancaster** [comment]
pushing to update coverage because it runs faster in the ☁️

I _think_ I still have one miss, but I'm not sure why because it looks like the tests should be going down that path

**Pull request merged**
